pkg/manifests/manifests.go

```diff
@@ -3581,3 +3713,44 @@ func trustedCABundleVolume(configMapName, volumeName string) v1.Volume {
 		},
 	}
 }
+
+func getAlertmanagerTLSSecretsName(alertmanagerConfigs []AdditionalAlertmanagerConfig) []string {
+
+	secretsName := []string{}
+	for _, alertmanagerConfig := range alertmanagerConfigs {
+		if alertmanagerConfig.TLSConfig.CA != nil {
+			secretsName = append(secretsName, alertmanagerConfig.TLSConfig.CA.Name)
+		}
+		if alertmanagerConfig.TLSConfig.Cert != nil {
+			secretsName = append(secretsName, alertmanagerConfig.TLSConfig.Cert.Name)
+		}
+		if alertmanagerConfig.TLSConfig.Key != nil {
+			secretsName = append(secretsName, alertmanagerConfig.TLSConfig.Cert.Name)
+		}
+		if alertmanagerConfig.BearerToken != nil {
+			secretsName = append(secretsName, alertmanagerConfig.BearerToken.Name)
+		}
+	}
+	return removeEmptyDuplicates(secretsName)
+}
+
+func removeEmptyDuplicates(elements []string) []string {
+	// Use map to record duplicates as we find them.
+	encountered := map[string]bool{}
+	result := []string{}
+
+	for v := range elements {
+		if encountered[elements[v]] == true {
+			// Do not add duplicate.
+		} else {
+			// Record this element as an encountered element.
+			encountered[elements[v]] = true
+			if elements[v] != "" {
```

Comment:
not sure why the value would be empty?

Reply:
I did not add logic to judge whether the `name` is empty or not. so it has chance to be empty value.